### PR TITLE
Feature/gh 2307

### PR DESCRIPTION
**Pull request by cmccrea23**
Move /images under /public and replace references with $GLOBALS variable

below added by bradymiller:
Up For Grabs demo for this PR is at : https://www.open-emr.org/wiki/index.php/Development_Demo#Zeta_-_Up_For_Grabs_Demo

### CODE REVIEW

File: custom/download_qrda.php [hunk 1]
```diff
@@ -150,10 +150,10 @@ function closeMe() {
         <th scope="col" class="multiDownload">
             <input type="checkbox" name="checkAll" id="checkAll"/>
             <div style="display:none" id=downloadAll>
-                <img class='downloadIcon' src='<?php echo $GLOBALS['webroot'];?>/images/downbtn.gif' onclick=downloadAllXML(); />
+                <img class="downloadIcon" src="<?php echo $GLOBALS['onclick=downloadAllXML()'];?>"/>
```
Comment (bradymiller) [inline custom/download_qrda.php]: this doesn't look right

File: custom/download_qrda.php [hunk 2]
```diff
@@ -150,10 +150,10 @@ function closeMe() {
...
             </div>
-            <div style='display:none' id=spinAll>;
-                <img src='<?php echo $GLOBALS['webroot'];?>/interface/pic/ajax-loader.gif'/>
+            <div style="display:none" id=spinAll>
+                <img src="<?php echo $GLOBALS['images_static_relative'];?>/interface/pic/ajax-loader.gif"/>
```
Comment (bradymiller) [inline custom/download_qrda.php]: this doesn't look right. Shouldn't there just be /ajax-loader.gif rather than /interface/pic/ajax-loader.gif ?

File: custom/download_qrda.php [hunk 3]
```diff
@@ -205,15 +205,15 @@ function closeMe() {
                 echo "</td>";
                 echo "<td align=center>";
                 echo "<div id=download" . attr($counter) . ">";
-                echo "<img class='downloadIcon' src='" . $GLOBALS['webroot'] . "/images/downbtn.gif' onclick=downloadXML(" . attr_js($counter) . ",1); />";
+                echo "<img class='downloadIcon' src='" . $GLOBALS['images_static_relative'] . "/downbtn.gif' onclick=downloadXML(" . attr_js($counter) . ",1); />";
                 echo "</div>";
                 echo "<div style='display:none' id=spin" . attr($counter) . ">";
-                echo "<img src='" . $GLOBALS['webroot'] . "/interface/pic/ajax-loader.gif'/>";
+                echo "<img src='" . $GLOBALS['images_static_relative'] . "/interface/pic/ajax-loader.gif'/>";
```
Comment (bradymiller) [inline custom/download_qrda.php]: this doesn't look right. Shouldn't there just be /ajax-loader.gif rather than /interface/pic/ajax-loader.gif ?

File: interface/forms/eye_mag/php/eye_mag_functions.php [hunk 4]
```diff
@@ -4048,21 +4048,6 @@ function report_header($pid, $direction = 'shell')
...
-                        echo "<img src='$practice_logo' align='left' style='width:150px;margin:0px 10px;'><br />\n";
-                    }
-                } else {
-                    $practice_logo = "$OE_SITE_DIR/images/practice_logo.gif";
-                    if (file_exists($practice_logo)) {
-                        echo "<img src='$practice_logo' align='left' style='width:100px;margin:0px 10px;'><br />\n";
-                    }
-                }
-            ?>
-            </td>
```
Comment (bradymiller) [inline interface/forms/eye_mag/php/eye_mag_functions.php]: This code needs to be added back

File: library/edihistory/test_edih_sftp_files.php [hunk 5]
```diff
@@ -168,9 +168,10 @@ function edih_disp_sftp_upload()
     // sftp file upload
     // imaginary form and POST values
     $str_html = '';
+
     if (isset($_POST['post_sftp'])) {
-        $la = (isset($_POST['post_sftp'])) ? filter_input(INPUT_POST, 'post_sftp', FILTER_DEFAULT) :  '';
-        $x12ptnr = (isset($_POST['sftp_select'])) ? filter_input(INPUT_POST, 'sftp_select', FILTER_DEFAULT) : '';
+        $la = (isset($_POST['post_sftp'])) ? filter_input(INPUT_POST, 'post_sftp', FILTER_DEFAULT) : ;
+        $x12ptnr = (isset($_POST['sftp_select'])) ? filter_input(INPUT_POST, 'sftp_select', FILTER_DEFAULT) :;
```
Comment (bradymiller) [inline library/edihistory/test_edih_sftp_files.php]: revert above code changes

**Comment by bradymiller** [comment]
hi @cmccrea23 , Just completed a code review. Looking forward to the next revision! -brady

File: custom/download_qrda.php (hunk 2 above)
Comment (cmccrea23) [reply custom/download_qrda.php]: The $GLOBALS variable leads to $web_root/public/images, but this gif is in interface/pics, not public/images. Wouldn't that lead to an incorrect path?

File: custom/download_qrda.php (hunk 1 above)
Comment (cmccrea23) [reply custom/download_qrda.php]: You're right, my bad. On another note, I searched for that function and couldn't find where it's defined or used anywhere else. Should I remove the onClick entirely? I fixed the image issue.

File: custom/download_qrda.php (hunk 3 above)
Comment (cmccrea23) [reply custom/download_qrda.php]: The $GLOBALS variable leads to $web_root/public/images, but this gif is in interface/pics, not public/images. Wouldn't that lead to an incorrect path?

File: interface/forms/eye_mag/php/eye_mag_functions.php (hunk 4 above)
Comment (cmccrea23) [reply interface/forms/eye_mag/php/eye_mag_functions.php]: I added it back :) That image doesn't exist, though. How should I fix this?

File: custom/download_qrda.php (hunk 2 above)
Comment (bradymiller) [reply custom/download_qrda.php]: Then would revert this line back to original:
```
<img src='<?php echo $GLOBALS['webroot'];?>/interface/pic/ajax-loader.gif'/>
```

File: custom/download_qrda.php (hunk 3 above)
Comment (bradymiller) [reply custom/download_qrda.php]: Then would revert this line back to original:
```
<img src='<?php echo $GLOBALS['webroot'];?>/interface/pic/ajax-loader.gif'/>
```

**Comment by cmccrea23** [comment]
> hi @cmccrea23 ,
> 
> Turns out I was able to remove all of library/css/images in addition to the query-ui-1.8.21.custom.css script after some code mods. See here for the commit that did this, which is now in the codebase:
> [0ed27f1](https://github.com/openemr/openemr/commit/0ed27f1c1ddb256381fe8a171b60ce0d97e0ffdc)
> 
> So, recommend reverting your most recent commit in your branch (and PR) to avoid bunch of merge conflicts in future:
> 
> ```
> git revert dd624eca4df4eae0a4da05e10e0c87888b9a8b86
> ```

Awesome! 👏 I've changed the $GLOBALS['webroot'] back for the relevant references in download_qrda.php. Also found that the `downloadAllXML()` function I found in line 153 still doesn't exist, but there is an active `downloadAllXML()`. Could that function be what was meant in line 153?

File: custom/download_qrda.php [hunk 6]
```diff
@@ -150,10 +150,10 @@ function closeMe() {
         <th scope="col" class="multiDownload">
             <input type="checkbox" name="checkAll" id="checkAll"/>
             <div style="display:none" id=downloadAll>
-                <img class='downloadIcon' src='<?php echo $GLOBALS['webroot'];?>/images/downbtn.gif' onclick=downloadAllXML(); />
+                <img class="downloadIcon" src="<?php echo $GLOBALS['webroot'];?>/downbtn.gif" onclick="downloadAllXML();" />
```
Comment (bradymiller) [inline custom/download_qrda.php]: This one has been migrated, so should use `$GLOBALS['images_static_relative']`:
```
<img class="downloadIcon" src="<?php echo $GLOBALS['images_static_relative']; ?>/downbtn.gif" onclick="downloadAllXML();" />
```

Comment (bradymiller) [reply custom/download_qrda.php]: Also agree with dropping the onclick there since that function does not exist:
```
<img class="downloadIcon" src="<?php echo $GLOBALS['images_static_relative']; ?>/downbtn.gif" />
```

File: custom/download_qrda.php [hunk 7]
```diff
@@ -205,15 +205,15 @@ function closeMe() {
                 echo "</td>";
                 echo "<td align=center>";
                 echo "<div id=download" . attr($counter) . ">";
-                echo "<img class='downloadIcon' src='" . $GLOBALS['webroot'] . "/images/downbtn.gif' onclick=downloadXML(" . attr_js($counter) . ",1); />";
+                echo "<img class='downloadIcon' src='" . $GLOBALS['images_static_relative'] . "/downbtn.gif' onclick=downloadXML()" . attr_js($counter) . ",1); />";
```
Comment (bradymiller) [inline custom/download_qrda.php]: There is an extra parenthesis `)` added here (after `downloadXML(` )

File: interface/modules/zend_modules/module/Installer/view/installer/installer/index.phtml [hunk 8]
```diff
@@ -32,7 +32,7 @@ $zendModDir 	= $GLOBALS['zendModDir'];
 $depObj 	= $this->dependencyObject;
 
 ?>
-<h1><img src="<?php echo $this->basePath()?>/images/module.png" class="nav-menu-img"><?php echo $this->escapeHtml($title); ?></h1>
+<h1><img src="<?php echo $GLOBALS['images_static_relative']; ?>/module.png" class="nav-menu-img"><?php echo $this->escapeHtml($title); ?></h1>
```
Comment (bradymiller) [inline interface/modules/zend_modules/module/Installer/view/installer/installer/index.phtml]: will need to revert above line. The zend_modules directory has its own assets/images (`$this->basePath()` is not going to the main images directory)

**Comment by bradymiller** [comment]
Following can be removed:
images/leftbtn.gif
images/module.png
images/rightbtn.gif
images/upbtn.gif

**Comment by cmccrea23** [comment]
I corrected the issues brought up above. My only question now is: what would you like me to do with interface/forms/eye_mag/php/eye_mag_functions.php starting at line 4051? The function there references an image that doesn't exist.

**Comment by bradymiller** [comment]
hi @cmccrea23 ,
Do you mean the practice_logo.gif script. If so, then would not touch that since those are scripts in sites/* which is where store configurable scripts/images. And wouldn't touch anything there.

**Comment by bradymiller** [comment]
missed b_edit.png fix in line 161 in library/custom_template/ajax_code.php
(will do a quick check of the others tomorrow)

**Comment by bradymiller** [comment]
Need to set the path here:
https://github.com/openemr/openemr/blob/master/library/classes/TreeMenu.php#L605
(this should cover all those places where things like `branch.src  = this.iconpath + '/branch.gif';` are called)

**Comment by bradymiller** [comment]
missed deleteBtn.png in library/custom_template/ajax_code.php (line 165) and library/edi.inc (line 542)

**Comment by bradymiller** [comment]
file3.png should be covered by the above TreeMenu.php change, but will need to test this (since code is hard to follow).

**Comment by bradymiller** [comment]
Missed folder.gif in interface/modules/zend_modules/public/css/jquery.treeview-1.4.1/jquery.treeview.css (line 71) and library/js/jquery.treeview-1.4.1/jquery.treeview.css (line 72)

**Comment by bradymiller** [comment]
Missed info.png in interface/main/calendar/modules/PostCalendar/pntemplates/default/user/ajax_search.html (line 108)

**Comment by bradymiller** [comment]
Missed sortdown.gif in interface/reports/patient_list_creation.php (lines 449 and 459)

**Comment by bradymiller** [comment]
Missed sortup.gif in interface/reports/patient_list_creation.php (lines 456)

**Comment by bradymiller** [comment]
Missed stock_search-16.png in templates/documents/general_queue.html (lines 32, 234, 240)

**Comment by bradymiller** [comment]
hi @cmccrea23 , Review complete. Getting close. After above is addressed in your next revision, will then plan testing.

**Comment by bradymiller** [comment]
Forgot to add. Can also remove the following 2 images:
check.png
loading.gif

**Comment by cmccrea23** [comment]
I think I used the $GLOBAL variable correctly here, but I've never worked with PHP before so I'm not certain. Would you mind taking another look? Same with setting the path in the TreeMenu, I'm not confident I did it correctly. Sorry in advance, and thanks for your help 🙏

File: interface/main/calendar/modules/PostCalendar/pntemplates/default/admin/submit.html [hunk 9]
```diff
@@ -237,7 +237,7 @@
 				<input type="text"	name="patient_name" size="15" value="[-$patient_value-]" readonly >
 				<input type="hidden" name="event_pid" value="[-$event_pid-]">
 				<a href="javascript:{}" onclick="top.restoreSession();var URL='[-$webroot-]/controller.php?patient_finder&find&form_id=add_event[\'event_pid\']&form_name=add_event[\'patient_name\']&pid=true'; window.open(URL, 'add_event', 'toolbar=0,scrollbars=1,location=0,statusbar=1,menubar=0,resizable=1,width=450,height=400,left = 425,top = 250');">
-					<img src="[-$webroot-]/images/stock_search-16.png" border="0"</a>
+					<img src="<?php echo $GLOBALS['images_static_relative']; ?>/stock_search-16.png" border="0" </a>
```
Comment (bradymiller) [inline interface/main/calendar/modules/PostCalendar/pntemplates/default/admin/submit.html]: This is within a smarty template, which always complicates things a bit. I will get back to this on later.
TODO

File: interface/main/calendar/modules/PostCalendar/pntemplates/default/user/ajax_search.html [hunk 10]
```diff
@@ -105,7 +105,7 @@ <h2>[-php-] xl('Searching for appointments','e'); [-/php-]</h2>
         echo "<td class='calsearch_provider'>".$event['provider_name'];
         $imgtitle = $provinfo['fname'] . " " . xl('contact info') . ":\n";
         $imgtitle .= $provinfo['phonew1']."\n".$provinfo['street']."\n".$provinfo['city']." ".$provinfo['state'];
-        echo " <img class'provinfo' src='".$GLOBALS['webroot']."/images/info.png' title=\"".$imgtitle."\" />";
+        echo " <img class'provinfo' src='". $GLOBALS['images_static_relative']/deleteBtn.png . "' title=\"".$imgtitle."\" />";
```
Comment (bradymiller) [inline interface/main/calendar/modules/PostCalendar/pntemplates/default/user/ajax_search.html]:
```
echo " <img class'provinfo' src='". $GLOBALS['images_static_relative'] . "/deleteBtn.png' title=\"".$imgtitle."\" />";
```

File: interface/reports/patient_list_creation.php [hunk 11]
```diff
@@ -446,17 +446,17 @@ function printForm(){
             }
 
             for ($i = 0; $i < count($sort); $i++) {
-                $sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols(" . attr_js($sort[$i]) . ",'asc');\" ><img src=\"../../images/sortdown.gif\" border=0 alt=\"".xla('Sort Up')."\"></a>";
+                $sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols(" . attr_js($sort[$i]) . ",'asc');\" ><img src='". $GLOBALS['images_static_relative']/sortdown.gif . "' border=0 alt=\"".xla('Sort Up')."\"></a>";
```
Comment (bradymiller) [inline interface/reports/patient_list_creation.php]:
```
$sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols(" . attr_js($sort[$i]) . ",'asc');\" ><img src='" .  $GLOBALS['images_static_relative'] . "/sortdown.gif' border=0 alt=\"".xla('Sort Up')."\"></a>";
```

File: interface/reports/patient_list_creation.php [hunk 12]
```diff
@@ -446,17 +446,17 @@ function printForm(){
...
             }
 
             for ($i = 0; $i < count($sort); $i++) {
                 if ($sortby == $sort[$i]) {
                     switch ($sortorder) {
                         case "asc":
-                            $sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols(" . attr_js($sortby) . ",'desc');\" ><img src=\"../../images/sortup.gif\" border=0 alt=\"".xla('Sort Up')."\"></a>";
+                            $sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols(" . attr_js($sortby) . ",'desc');\" ><img src='". $GLOBALS['images_static_relative']/sortup.gif . "' border=0 alt=\"".xla('Sort Up')."\"></a>";
```
Comment (bradymiller) [inline interface/reports/patient_list_creation.php]:
```
$sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols(" . attr_js($sortby) . ",'desc');\" ><img src='" .  $GLOBALS['images_static_relative'] . "/sortup.gif' border=0 alt=\"".xla('Sort Up')."\"></a>";
```

File: interface/reports/patient_list_creation.php [hunk 13]
```diff
@@ -446,17 +446,17 @@ function printForm(){
...
                             break;
                         case "desc":
-                            $sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols('" . attr_js($sortby) . "','asc');\" onclick=\"top.restoreSession()\"><img src=\"../../images/sortdown.gif\" border=0 alt=\"".xla('Sort Down')."\"></a>";
+                            $sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols('" . attr_js($sortby) . "','asc');\" onclick=\"top.restoreSession()\"><img src='". $GLOBALS['images_static_relative']/sortdown.gif . "' border=0 alt=\"".xla('Sort Down')."\"></a>";
```
Comment (bradymiller) [inline interface/reports/patient_list_creation.php]:
```
$sortlink[$i] = "<a href=\"#\" onclick=\"sortingCols('" . attr_js($sortby) . "','asc');\" onclick=\"top.restoreSession()\"><img src='" . $GLOBALS['images_static_relative'] . "/sortdown.gif' border=0 alt=\"".xla('Sort Down')."\"></a>";
```

File: library/classes/TreeMenu.php [hunk 14]
```diff
@@ -602,7 +602,7 @@ function __construct(&$structure, $options = array(), $isDynamic = true)
         $this->isDynamic = $isDynamic;
 
         // Defaults
-        $this->images           = 'images';
+        $this->images           = $GLOBALS['images_static_relative'];
```
Comment (bradymiller) [inline library/classes/TreeMenu.php]: Will need to hard code this to:
```
$this->images           = 'public/images';
```

File: library/custom_template/ajax_code.php [hunk 15]
```diff
@@ -158,11 +158,11 @@
         $i++;
         echo "<li id='clorder_".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."' style='cursor:pointer'><span>";
         if (acl_check('nationnotes', 'nn_configure')) {
-            echo "<img src='../../images/b_edit.png' onclick=update_item_div('".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."')>";
+            echo "<img src='". $GLOBALS['images_static_relative']/deleteBtn.png . "' onclick=update_item_div('".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."')>";
```
Comment (bradymiller) [inline library/custom_template/ajax_code.php]:
```
echo "<img src='" . $GLOBALS['images_static_relative'] . "/deleteBtn.png' onclick=update_item_div('".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."')>";
```

File: library/custom_template/ajax_code.php [hunk 16]
```diff
@@ -158,11 +158,11 @@
...
         }
         echo "<div style='display:inline' id='".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."' onclick=\"moveOptions_11('".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."', 'textarea1');\">".htmlspecialchars($row['cl_list_item_long'], ENT_QUOTES)."</div>";
         if (acl_check('nationnotes', 'nn_configure')) {
-            echo "<img src='../../images/deleteBtn.png' onclick=\"delete_item('".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."')\">";
+            echo "<img src='". $GLOBALS['images_static_relative']/deleteBtn.png . "' onclick=\"delete_item('".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."')\">";
```
Comment (bradymiller) [inline library/custom_template/ajax_code.php]:
```
echo "<img src='" . $GLOBALS['images_static_relative'] . "/deleteBtn.png' onclick=\"delete_item('".htmlspecialchars($row['cl_list_slno'], ENT_QUOTES)."')\">";
```

File: library/edi.inc [hunk 17]
```diff
@@ -539,7 +539,7 @@ function show_elig($res, $X12info, $segTer, $compEleSep)
 				<td class ='detail' style='width:6%;'>".  htmlspecialchars($row['subscriber_sex'], ENT_NOQUOTES) ."</td>
 				<td class ='detail' style='width:9%;'>".  htmlspecialchars($row['subscriber_ss'], ENT_NOQUOTES) ."</td>
 				<td class ='detail' style='width:2%;'>
-				<img src='../../images/deleteBtn.png' title=' .htmlspecialchars( xl('Delete Row'), ENT_QUOTES) . ' style='cursor:pointer;cursor:hand;' onclick='deletetherow(\"" . $i."_". htmlspecialchars($row['policy_number'], ENT_QUOTES) . "\")'>
+				<img src='". $GLOBALS['images_static_relative']/deleteBtn.png . "' title=' .htmlspecialchars( xl('Delete Row'), ENT_QUOTES) . ' style='cursor:pointer;cursor:hand;' onclick='deletetherow(\"" . $i."_". htmlspecialchars($row['policy_number'], ENT_QUOTES) . "\")'>
```
Comment (bradymiller) [inline library/edi.inc]:
```
<img src='" . $GLOBALS['images_static_relative'] . "/deleteBtn.png' title=' .htmlspecialchars( xl('Delete Row'), ENT_QUOTES) . ' style='cursor:pointer;cursor:hand;' onclick='deletetherow(\"" . $i."_". htmlspecialchars($row['policy_number'], ENT_QUOTES) . "\")'>
```

**Comment by bradymiller** [comment]
hi @cmccrea23 ,
Getting very close. Left more instructions above. I'll show how to make this work in the smarty template script (item with TODO in my post) tomorrow.

File: interface/main/calendar/modules/PostCalendar/pntemplates/default/admin/submit.html (hunk 9 above)
Comment (bradymiller) [reply interface/main/calendar/modules/PostCalendar/pntemplates/default/admin/submit.html]: ok, lets try the following here:
```
<img src="[-php-] echo $GLOBALS['images_static_relative']; [-/php-]/stock_search-16.png" border="0" </a>
```

**Comment by cmccrea23** [comment]
Sorry this took so long for me to do. I just started a new job at the beginning of the month and haven't had much time for outside projects 😣hope this fixes it!

File: interface/main/calendar/modules/PostCalendar/pntemplates/default/user/ajax_search.html [hunk 18]
```diff
@@ -105,7 +105,7 @@ <h2>[-php-] xl('Searching for appointments','e'); [-/php-]</h2>
         echo "<td class='calsearch_provider'>".$event['provider_name'];
         $imgtitle = $provinfo['fname'] . " " . xl('contact info') . ":\n";
         $imgtitle .= $provinfo['phonew1']."\n".$provinfo['street']."\n".$provinfo['city']." ".$provinfo['state'];
-        echo " <img class'provinfo' src='". $GLOBALS['images_static_relative']/deleteBtn.png . "' title=\"".$imgtitle."\" />";
+        echo " <img class'provinfo' src='". $GLOBALS[' images_static_relative'] . "/deleteBtn.png' title=\"".$imgtitle."\" />";
```
Comment (bradymiller) [inline interface/main/calendar/modules/PostCalendar/pntemplates/default/user/ajax_search.html]: There is a space between the ' and i characters here

**Comment by bradymiller** [comment]
hi @cmccrea23 , no prob on the delay and congrats on the job! tortoise always wins the race :)
I only noted one more minor issue on code review. When that's addressed, then I'll do testing and last review before bringing it into the codebase. thanks! -brady

**Comment by bradymiller** [comment]
hi @cmccrea23 ,
Very nice work on this large project!
Did testing and made some fixes, which brought into this PR:
https://github.com/openemr/openemr/pull/2310/commits/1fab949a0c70eb4ad4c93810a6451cf84f18168e
Going to bring this PR into the main codebase now. Thanks for the code contribution!
-brady
